cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -593,6 +603,184 @@ private Future<List<Condition>> maybeRestartConnectorTask(Reconciliation reconci
+            }
+            case reset -> {
+                return resetConnectorOffsets(reconciliation, host, apiClient, connectorName, resource, conditions);
+            }
+            default -> {
+                return Future.succeededFuture(conditions);
+            }
+        }
+    }
+
+    /**
+     * Fetches the connector offsets and writes them to a ConfigMap.
+     * This operation requires the listOffsets property to be set on
+     * the Custom Resource giving the name of the ConfigMap to use.
+     */
+    @SuppressWarnings({ "rawtypes" })
+    Future<List<Condition>> listConnectorOffsets(Reconciliation reconciliation, String host, KafkaConnectApi apiClient, String connectorName, CustomResource resource, KafkaConnectorSpec connectorSpec, List<Condition> conditions) {
+        Optional<ListOffsets> listOffsetsConfig = Optional.ofNullable(connectorSpec.getListOffsets());
+        if (listOffsetsConfig.isEmpty()) {
+            String message = String.format("Failed to list the connector offsets due to missing property listOffsets in %s CR.", resource.getKind());
+            LOGGER.warnCr(reconciliation, message);
+            conditions.add(StatusUtils.buildWarningCondition("ListOffsets", message));
+            return Future.succeededFuture(conditions);
+        }
+        String configMapName = connectorSpec.getListOffsets().getConfigMapReference().getName();
+        return apiClient.getConnectorOffsets(reconciliation, host, port, connectorName)
+                .compose(offsets -> {
+                    Map<String, String> data = new HashMap<>(1);
+                    data.put(getConnectorOffsetsConfigMapEntryKey(connectorName), offsets);
+                    return generateListOffsetsConfigMap(configMapName, resource, data);
+                })
+                .compose(configMap -> configMapOperations.reconcile(reconciliation, resource.getMetadata().getNamespace(), configMapName, configMap))
+                .compose(v -> removeConnectorOffsetsAnnotation(reconciliation, resource))
+                .compose(v -> Future.succeededFuture(conditions), throwable -> {
+                    // Don't fail reconciliation on error from listing offsets - add a warning and repeat list on next reconcile
+                    String message = "Encountered error listing connector offsets. " + throwable.getMessage();
+                    LOGGER.warnCr(reconciliation, message);
+                    conditions.add(StatusUtils.buildWarningCondition("ListOffsets", message));
+                    return Future.succeededFuture(conditions);
+                });
+    }
+
+
+
+    /**
+     * Generates a connector offsets ConfigMap with the provided data.
+     * If there is already an existing ConfigMap it includes the existing labels and data, and uses the existing owner references (if present).
+     *
+     * @return The generated ConfigMap
```

Comment:
Here as well as in the other methods here. If you do the Javadocs, you should do them properly.

Reply:
Good point, I had added a basic comment as I was writing the methods and meant to put the proper Javadoc in later, but obviously forgot. All the Javadoc should be correct now